### PR TITLE
sort out generated binaries into proper names

### LINKED ISSUE #1: change `make dist` to match actual output filenames
`make dist` should generate files that match  https://ssl.icu-project.org/files/icu4c/60.2/ (including the shasums).

### PR DESCRIPTION
**Pull request by srl295**
may not work for all versions, etc. But it is a start.

Fixes: https://github.com/unicode-org/icu-docker/issues/1

### CODE REVIEW

**Comment by srl295** [comment]
before and after

```
# before
icu-rrelease-66-1-x86_64-pc-linux-gnu-Fedora-31.tgz             icu-rrelease-66-1-x86_64-pc-linux-gnu-Ubuntu-18.04-src.d
icu-rrelease-66-1-x86_64-pc-linux-gnu-Fedora-31.tgz.md5         icu-rrelease-66-1-x86_64-pc-linux-gnu-Ubuntu-18.04.tgz
icu-rrelease-66-1-x86_64-pc-linux-gnu-Ubuntu-18.04-sdoc.tgz     icu-rrelease-66-1-x86_64-pc-linux-gnu-Ubuntu-18.04.tgz.md5
```

```
#after
icu4c-66.1/SOURCEDOC-icu4c-66_1-SOURCEDOC.tgz   icu4c-66.1/icu4c-66_1-data.zip                  icu4c-66.1/icu4c-66_1-src.zip
icu4c-66.1/icu4c-66_1-Fedora31-x64.tgz          icu4c-66.1/icu4c-66_1-docs.zip
icu4c-66.1/icu4c-66_1-Ubuntu18.04-x64.tgz       icu4c-66.1/icu4c-66_1-src.tgz

```

Oooh! Aaaah!

File: sort-out-dist.sh [hunk 1]
```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+# Sort out the /dist folder, which is somewhat of a mess.
+# Ideally `make dist` would do this. But until then.
+cd ./dist || exit 1
```
Comment (roubert) [inline sort-out-dist.sh:5]: The `./` prefix doesn't actually do or mean anything in this context.

File: sort-out-dist.sh [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+# now get a list of stuff
+STUFF=$(find icu-* -type f \( -name '*.zip' -o -name '*.tgz' \))
+
+trymove()
+{
+    base2=$(basename $1)
+    if [ -f ${2}/${3} ];
```
Comment (roubert) [inline sort-out-dist.sh:13]: The `;` suffix doesn't actually do or mean anything in this context.

File: sort-out-dist.sh [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    fi
+}
+
+for file in ${STUFF};
+do
+    #echo "# ${file}"
+    base=$(basename $file)
+    # rver is '66-1'
+    rver=$(echo $file | grep "^icu-rrelease-[0-9][0-9]-[0-9]" | cut -d- -f3-4)
+    if [ "${rver}" = "" ];
```
Comment (roubert) [inline sort-out-dist.sh:28]: The `;` suffix doesn't actually do or mean anything in this context.

File: sort-out-dist.sh [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    then
+        echo "# ${file} - could not extract release version."
+        continue
+    fi
+    uver=$(echo $rver | tr - _)
+    prefix=icu4c-${uver}
+    dir=icu4c-$(echo $rver | tr - .)
+    mkdir -pv ./${dir}/
```
Comment (roubert) [inline sort-out-dist.sh:36]: Neither the `./` prefix nor the `/` suffix actually do or mean anything in this context.

File: sort-out-dist.sh [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+            ;;
+        icu4c-${uver}-*)
+            # ignore anything else with uver
+            echo "# Ignored: ${base}"
+            ;;
+        *):
+            echo "## Warn: could not classify ${base}"
+            ;;
+    esac
+done
```
Comment (roubert) [inline sort-out-dist.sh:68]: No newline at end of file.

File: sort-out-dist.sh [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+            ;;
+        ${prefix}-docs.zip|${prefix}-src.tgz|${prefix}-src.zip|${prefix}-data.zip)
+            # already has the right name
+            trymove ${file} ${dir} ${base}
+            ;;
+        icu4c-${uver}-*)
+            # ignore anything else with uver
+            echo "# Ignored: ${base}"
+            ;;
+        *):
```
Comment (roubert) [inline sort-out-dist.sh:64]: Is this `:` a typo? (If not, what does it do?)

**Review by roubert — APPROVED** [review]
LGTM

**Pull request merged**